papergres.go

```diff
@@ -533,22 +545,27 @@ func goToSQLName(name string) string {
 }
 
 // insertArgs creates the insert arg slice for an object
-func insertArgs(obj interface{}) []interface{} {
-	final, _ := prepareFields(obj)
+func insertArgs(obj interface{}, withPK bool) []interface{} {
+	final, _ := prepareFields(obj, withPK)
 	args := make([]interface{}, len(final))
 	for i, f := range final {
 		args[i] = f.Value
 	}
 	return args
 }
 
-// prepareFields performs necessary transformations for the insert statement
-func prepareFields(obj interface{}) (nfields []*Field, primary *Field) {
+// prepareFields performs necessary transformations for the insert statement.
+// If `withPK` is false: It does not account a primary key Field to list of
+// fields to append, else, primary key is also considered.
+func prepareFields(obj interface{}, withPK bool) (nfields []*Field, primary *Field) {
```

Comment:
See comment about passing around the same variable multiple times.

Reply:
Value semantics are the idiomatic way to Go :)
Read the above comment for a detailed explanation on how the Go compiler works differently and uses escape analysis to make passing by value a much cheaper call.

Reply:
Also, the withPK is just a control flag. so it's safe to pass it around to internal functions which determine the eventual output.